src/librustc_mir/transform/qualify_consts.rs

```diff
@@ -135,6 +145,12 @@ enum ValueSource<'a, 'tcx> {
     },
 }
 
+/// A "qualif" is a way to lookg for something "bad" in the MIR that would prevent
+/// proper const evaluation.  So `return true` means "I found something bad, no reason
```

Comment:
This seems... overly simplistic. And inaccurate - it doesn't have to prevent const evaluation!
What it's about is "disqualifying from promotion".

Also, `s/  / /`.

Reply:
Well, `NeedsDrop` is not just used for promotion.